sound/soc/sof/utils.c

```diff
@@ -66,3 +66,20 @@ int sof_create_platform_device(struct sof_platform_priv *priv)
 	return 0;
 }
 EXPORT_SYMBOL(sof_create_platform_device);
+
+int sof_boot_wait_timeout(struct snd_sof_dev *sdev)
+{
+	int ret;
+
+	ret = wait_event_timeout(sdev->boot_wait, sdev->boot_complete,
+				 msecs_to_jiffies(sdev->boot_timeout));
+	if (ret == 0) {
+		dev_err(sdev->dev, "error: firmware boot timeout\n");
+		return -EIO;
```

Comment:
return ret, also best to print ret value in dev_err too.

Reply:
not relevant anymore with recent updates.